viz/cmd/root.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"fmt"
+	"regexp"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultLinkerdNamespace = "linkerd"
+	defaultVizNamespace     = "linkerd-viz"
+)
+
+var (
+	apiAddr               string // An empty value means "use the Kubernetes configuration"
+	controlPlaneNamespace string
+	namespace             string
+	kubeconfigPath        string
+	kubeContext           string
+	impersonate           string
+	impersonateGroup      []string
+	verbose               bool
+
+	// These regexs are not as strict as they could be, but are a quick and dirty
+	// sanity check against illegal characters.
+	alphaNumDash = regexp.MustCompile(`^[a-zA-Z0-9-]+$`)
+)
+
+// NewCmdViz returns a new jeager command
+func NewCmdViz() *cobra.Command {
+	jaegerCmd := &cobra.Command{
+		Use:   "viz",
+		Short: "viz manages the linkerd-viz extension of Linkerd service mesh",
+		Long:  `viz manages the linkerd-viz extension of Linkerd service mesh.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// enable / disable logging
+			if verbose {
+				log.SetLevel(log.DebugLevel)
+			} else {
+				log.SetLevel(log.PanicLevel)
+			}
+
+			if !alphaNumDash.MatchString(controlPlaneNamespace) {
+				return fmt.Errorf("%s is not a valid namespace", controlPlaneNamespace)
+			}
+
+			return nil
+		},
+	}
+
+	jaegerCmd.PersistentFlags().StringVarP(&controlPlaneNamespace, "linkerd-namespace", "L", defaultLinkerdNamespace, "Namespace in which Linkerd is installed")
+	jaegerCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", defaultVizNamespace, "Namespace in which Jaeger extension is installed")
```

Comment:
As a matter of fact, do we have to declare all these root commands here again given they're already declared in the parent command's root file?

Reply:
Hmm, Interesting. I initially declared them because we were going with the separate command way of doing it, but now its just a subcommand.

They are currently not accessible because they are not public, and these are in different pkgs. Should we try doing that or keep them separate? 🤔

Reply:
Per my suggestions above, I recommend getting rid of the `linkerd-viz-namespace` flag in the main control plane altogether.

Reply:
Removed `linkerdVizNamespace` from the main control-plane and its helm chart.